view/rooms/app.js

```diff
@@ -1071,7 +1071,7 @@ function buildRedirectUrl(url, roomId) {
   return redirectUrl;
 }
 
-window.onbeforeunload = confirmUserExit;
+window.addEventListener('beforeunload', confirmUserExit);
```

Comment:
is this where we generate the event "user has left the room" or here is where the pop message is generated?

Reply:
it happens inside the function `confirmUserExit` which as agreed I refactored to be called`generateUserExitEvent`